src/app/header/header.component.ts

```diff
@@ -1,16 +1,67 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { UserService } from '../shared/user.service';
+import { UserDetails } from '../header/UserDetails';
+import { Cart } from '../cart/cart';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styles: [
   ]
 })
+
 export class HeaderComponent implements OnInit {
 
-  constructor() { }
+  userDetails: UserDetails = new UserDetails();
+  @ViewChild('cartItemsCount', { static: true }) cartItemsCountIcon: ElementRef;
+  count: number = 0;
+  flag: boolean = false;
+
+  constructor(private router: Router, private service: UserService, private location: Location) {
+  }
 
   ngOnInit(): void {
+    this.show();
+  }
+
+  ngAfterContentChecked(): void {
+    this.updateCartCount();
   }
 
+  updateCartCount(): void {
+    const localStoredCartItems: Cart[] = JSON.parse(localStorage.getItem("cartItems"));
+    if (localStoredCartItems.length > 0) {
+      this.cartItemsCountIcon.nativeElement.classList.remove('mat-badge-hidden');
+      this.count = localStoredCartItems.length;
```

Comment:
I am guessing this.count is the count of the cart items right??

Please follow proper naming conventions

Reply:
yes that change was committed in another branch my mistake my bad will update here as well